// @prismicio/react/dist/react-server/PrismicRichText.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const jsxRuntime = require("react/jsx-runtime");
const React = require("react");
const prismicR = require("@prismicio/richtext");
const PrismicLink = require("./PrismicLink.cjs");
const devMsg = require("../lib/devMsg.cjs");
const isFilled = require('./../_node_modules/@prismicio/client/dist/helpers/isFilled.cjs');
function _interopNamespaceDefault(e) {
  const n = Object.create(null, { [Symbol.toStringTag]: { value: "Module" } });
  if (e) {
    for (const k in e) {
      if (k !== "default") {
        const d = Object.getOwnPropertyDescriptor(e, k);
        Object.defineProperty(n, k, d.get ? d : {
          enumerable: true,
          get: () => e[k]
        });
      }
    }
  }
  n.default = e;
  return Object.freeze(n);
}
const React__namespace = /* @__PURE__ */ _interopNamespaceDefault(React);
const prismicR__namespace = /* @__PURE__ */ _interopNamespaceDefault(prismicR);
const createDefaultSerializer = (args) => prismicR__namespace.wrapMapSerializer({
  heading1: ({ children, key }) => /* @__PURE__ */ jsxRuntime.jsx("h1", { children }, key),
  heading2: ({ children, key }) => /* @__PURE__ */ jsxRuntime.jsx("h2", { children }, key),
  heading3: ({ children, key }) => /* @__PURE__ */ jsxRuntime.jsx("h3", { children }, key),
  heading4: ({ children, key }) => /* @__PURE__ */ jsxRuntime.jsx("h4", { children }, key),
  heading5: ({ children, key }) => /* @__PURE__ */ jsxRuntime.jsx("h5", { children }, key),
  heading6: ({ children, key }) => /* @__PURE__ */ jsxRuntime.jsx("h6", { children }, key),
  paragraph: ({ children, key }) => /* @__PURE__ */ jsxRuntime.jsx("p", { children }, key),
  preformatted: ({ node, key }) => /* @__PURE__ */ jsxRuntime.jsx("pre", { children: node.text }, key),
  strong: ({ children, key }) => /* @__PURE__ */ jsxRuntime.jsx("strong", { children }, key),
  em: ({ children, key }) => /* @__PURE__ */ jsxRuntime.jsx("em", { children }, key),
  listItem: ({ children, key }) => /* @__PURE__ */ jsxRuntime.jsx("li", { children }, key),
  oListItem: ({ children, key }) => /* @__PURE__ */ jsxRuntime.jsx("li", { children }, key),
  list: ({ children, key }) => /* @__PURE__ */ jsxRuntime.jsx("ul", { children }, key),
  oList: ({ children, key }) => /* @__PURE__ */ jsxRuntime.jsx("ol", { children }, key),
  image: ({ node, key }) => {
    const img = /* @__PURE__ */ jsxRuntime.jsx("img", { src: node.url, alt: node.alt ?? void 0, "data-copyright": node.copyright ? node.copyright : void 0 });
    return /* @__PURE__ */ jsxRuntime.jsx("p", { className: "block-img", children: node.linkTo ? /* @__PURE__ */ jsxRuntime.jsx(PrismicLink.PrismicLink, { linkResolver: args.linkResolver, internalComponent: args.internalLinkComponent, externalComponent: args.externalLinkComponent, field: node.linkTo, children: img }) : img }, key);
  },
  embed: ({ node, key }) => /* @__PURE__ */ jsxRuntime.jsx("div", { "data-oembed": node.oembed.embed_url, "data-oembed-type": node.oembed.type, "data-oembed-provider": node.oembed.provider_name, dangerouslySetInnerHTML: { __html: node.oembed.html ?? "" } }, key),
  hyperlink: ({ node, children, key }) => /* @__PURE__ */ jsxRuntime.jsx(PrismicLink.PrismicLink, { field: node.data, linkResolver: args.linkResolver, internalComponent: args.internalLinkComponent, externalComponent: args.externalLinkComponent, children }, key),
  label: ({ node, children, key }) => /* @__PURE__ */ jsxRuntime.jsx("span", { className: node.data.label, children }, key),
  span: ({ text, key }) => {
    const result = [];
    let i = 0;
    for (const line of text.split("\n")) {
      if (i > 0) {
        result.push(/* @__PURE__ */ jsxRuntime.jsx("br", {}, `${i}__break`));
      }
      result.push(/* @__PURE__ */ jsxRuntime.jsx(React__namespace.Fragment, { children: line }, `${i}__line`));
      i++;
    }
    return /* @__PURE__ */ jsxRuntime.jsx(React__namespace.Fragment, { children: result }, key);
  }
});
function PrismicRichText({ linkResolver, field, fallback, components, externalLinkComponent, internalLinkComponent, ...restProps }) {
  return React__namespace.useMemo(() => {
    if (typeof process !== "undefined" && process.env.NODE_ENV === "development") {
      if ("className" in restProps) {
        console.warn(`[PrismicRichText] className cannot be passed to <PrismicRichText> since it renders an array without a wrapping component. For more details, see ${devMsg.devMsg("classname-is-not-a-valid-prop")}.`, field);
      }
    }
    if (isFilled.richText(field)) {
      const serializer = prismicR__namespace.composeSerializers(typeof components === "object" ? prismicR__namespace.wrapMapSerializer(components) : components, createDefaultSerializer({
        linkResolver,
        internalLinkComponent,
        externalLinkComponent
      }));
      const serialized = prismicR__namespace.serialize(field, (type, node, text, children, key) => {
        const result = serializer(type, node, text, children, key);
        if (React__namespace.isValidElement(result) && result.key == null) {
          return React__namespace.cloneElement(result, { key });
        } else {
          return result;
        }
      });
      return /* @__PURE__ */ jsxRuntime.jsx(jsxRuntime.Fragment, { children: serialized });
    } else {
      return fallback != null ? /* @__PURE__ */ jsxRuntime.jsx(jsxRuntime.Fragment, { children: fallback }) : null;
    }
  }, [
    field,
    internalLinkComponent,
    externalLinkComponent,
    components,
    linkResolver,
    fallback
  ]);
}
exports.PrismicRichText = PrismicRichText;
