// @prismicio/react/dist/clientHooks.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const useStatefulPrismicClientMethod = require("./useStatefulPrismicClientMethod.cjs");
const usePrismicDocuments = (...args) => useStatefulPrismicClientMethod.useStatefulPrismicClientMethod("get", args);
const useFirstPrismicDocument = (...args) => useStatefulPrismicClientMethod.useStatefulPrismicClientMethod("getFirst", args);
const useAllPrismicDocumentsDangerously = (...args) => useStatefulPrismicClientMethod.useStatefulPrismicClientMethod("dangerouslyGetAll", args);
const usePrismicDocumentByID = (...args) => useStatefulPrismicClientMethod.useStatefulPrismicClientMethod("getByID", args);
const usePrismicDocumentsByIDs = (...args) => useStatefulPrismicClientMethod.useStatefulPrismicClientMethod("getByIDs", args);
const useAllPrismicDocumentsByIDs = (...args) => useStatefulPrismicClientMethod.useStatefulPrismicClientMethod("getAllByIDs", args);
const usePrismicDocumentByUID = (...args) => useStatefulPrismicClientMethod.useStatefulPrismicClientMethod("getByUID", args);
const usePrismicDocumentsByUIDs = (...args) => useStatefulPrismicClientMethod.useStatefulPrismicClientMethod("getByUIDs", args);
const useAllPrismicDocumentsByUIDs = (...args) => useStatefulPrismicClientMethod.useStatefulPrismicClientMethod("getAllByUIDs", args);
const useSinglePrismicDocument = (...args) => useStatefulPrismicClientMethod.useStatefulPrismicClientMethod("getSingle", args);
const usePrismicDocumentsByType = (...args) => useStatefulPrismicClientMethod.useStatefulPrismicClientMethod("getByType", args);
const useAllPrismicDocumentsByType = (...args) => useStatefulPrismicClientMethod.useStatefulPrismicClientMethod("getAllByType", args);
const usePrismicDocumentsByTag = (...args) => useStatefulPrismicClientMethod.useStatefulPrismicClientMethod("getByTag", args);
const useAllPrismicDocumentsByTag = (...args) => useStatefulPrismicClientMethod.useStatefulPrismicClientMethod("getAllByTag", args);
const usePrismicDocumentsBySomeTags = (...args) => useStatefulPrismicClientMethod.useStatefulPrismicClientMethod("getBySomeTags", args);
const useAllPrismicDocumentsBySomeTags = (...args) => useStatefulPrismicClientMethod.useStatefulPrismicClientMethod("getAllBySomeTags", args);
const usePrismicDocumentsByEveryTag = (...args) => useStatefulPrismicClientMethod.useStatefulPrismicClientMethod("getByEveryTag", args);
const useAllPrismicDocumentsByEveryTag = (...args) => useStatefulPrismicClientMethod.useStatefulPrismicClientMethod("getAllByEveryTag", args);
exports.useAllPrismicDocumentsByEveryTag = useAllPrismicDocumentsByEveryTag;
exports.useAllPrismicDocumentsByIDs = useAllPrismicDocumentsByIDs;
exports.useAllPrismicDocumentsBySomeTags = useAllPrismicDocumentsBySomeTags;
exports.useAllPrismicDocumentsByTag = useAllPrismicDocumentsByTag;
exports.useAllPrismicDocumentsByType = useAllPrismicDocumentsByType;
exports.useAllPrismicDocumentsByUIDs = useAllPrismicDocumentsByUIDs;
exports.useAllPrismicDocumentsDangerously = useAllPrismicDocumentsDangerously;
exports.useFirstPrismicDocument = useFirstPrismicDocument;
exports.usePrismicDocumentByID = usePrismicDocumentByID;
exports.usePrismicDocumentByUID = usePrismicDocumentByUID;
exports.usePrismicDocuments = usePrismicDocuments;
exports.usePrismicDocumentsByEveryTag = usePrismicDocumentsByEveryTag;
exports.usePrismicDocumentsByIDs = usePrismicDocumentsByIDs;
exports.usePrismicDocumentsBySomeTags = usePrismicDocumentsBySomeTags;
exports.usePrismicDocumentsByTag = usePrismicDocumentsByTag;
exports.usePrismicDocumentsByType = usePrismicDocumentsByType;
exports.usePrismicDocumentsByUIDs = usePrismicDocumentsByUIDs;
exports.useSinglePrismicDocument = useSinglePrismicDocument;
