// @prismicio/react/dist/react-server.cjs
"use strict";
Object.defineProperty(exports, Symbol.toStringTag, { value: "Module" });
const PrismicLink = require("./react-server/PrismicLink.cjs");
const PrismicRichText = require("./react-server/PrismicRichText.cjs");
const unsupported = require("./react-server/unsupported.cjs");
const PrismicText = require("./PrismicText.cjs");
const prismicR = require("@prismicio/richtext");
const PrismicImage = require("./PrismicImage.cjs");
const SliceZone = require("./SliceZone.cjs");
const PrismicToolbar = require("./PrismicToolbar.cjs");
exports.PrismicLink = PrismicLink.PrismicLink;
exports.PrismicRichText = PrismicRichText.PrismicRichText;
exports.PrismicProvider = unsupported.PrismicProvider;
exports.useAllPrismicDocumentsByEveryTag = unsupported.useAllPrismicDocumentsByEveryTag;
exports.useAllPrismicDocumentsByIDs = unsupported.useAllPrismicDocumentsByIDs;
exports.useAllPrismicDocumentsBySomeTags = unsupported.useAllPrismicDocumentsBySomeTags;
exports.useAllPrismicDocumentsByTag = unsupported.useAllPrismicDocumentsByTag;
exports.useAllPrismicDocumentsByType = unsupported.useAllPrismicDocumentsByType;
exports.useAllPrismicDocumentsByUIDs = unsupported.useAllPrismicDocumentsByUIDs;
exports.useAllPrismicDocumentsDangerously = unsupported.useAllPrismicDocumentsDangerously;
exports.useFirstPrismicDocument = unsupported.useFirstPrismicDocument;
exports.usePrismicClient = unsupported.usePrismicClient;
exports.usePrismicContext = unsupported.usePrismicContext;
exports.usePrismicDocumentByID = unsupported.usePrismicDocumentByID;
exports.usePrismicDocumentByUID = unsupported.usePrismicDocumentByUID;
exports.usePrismicDocuments = unsupported.usePrismicDocuments;
exports.usePrismicDocumentsByEveryTag = unsupported.usePrismicDocumentsByEveryTag;
exports.usePrismicDocumentsByIDs = unsupported.usePrismicDocumentsByIDs;
exports.usePrismicDocumentsBySomeTags = unsupported.usePrismicDocumentsBySomeTags;
exports.usePrismicDocumentsByTag = unsupported.usePrismicDocumentsByTag;
exports.usePrismicDocumentsByType = unsupported.usePrismicDocumentsByType;
exports.usePrismicDocumentsByUIDs = unsupported.usePrismicDocumentsByUIDs;
exports.usePrismicPreviewResolver = unsupported.usePrismicPreviewResolver;
exports.useSinglePrismicDocument = unsupported.useSinglePrismicDocument;
exports.PrismicText = PrismicText.PrismicText;
Object.defineProperty(exports, "Element", {
  enumerable: true,
  get: () => prismicR.Element
});
exports.PrismicImage = PrismicImage.PrismicImage;
exports.SliceZone = SliceZone.SliceZone;
exports.TODOSliceComponent = SliceZone.TODOSliceComponent;
exports.PrismicToolbar = PrismicToolbar.PrismicToolbar;
